### PR TITLE
Additional support for vector indexes.

### PR DESCRIPTION
**Pull request by nicktobey**
This PR adds the vector function VEC_DISTANCE to GMS, as well as support for adding and altering vector indexes, and support for `SHOW CREATE` on tables with vector indexes.

Vector indexes are not yet supported in Dolt. The corresponding version bump in Dolt will have checks preventing vector indexes from being added to Dolt tables.

### CODE REVIEW

**Comment by nicktobey** [comment]
Dolt bump here: https://github.com/dolthub/dolt/pull/8516

File: sql/sqlfmt.go [hunk 1]
```diff
@@ -102,7 +102,7 @@ func GenerateCreateTablePrimaryKeyDefinition(pkCols []string) string {
 
 // GenerateCreateTableIndexDefinition returns index definition string for 'CREATE TABLE' statement
 // for given index. This part comes after primary key definition if there is any.
-func GenerateCreateTableIndexDefinition(isUnique, isSpatial, isFullText bool, indexID string, indexCols []string, comment string) string {
+func GenerateCreateTableIndexDefinition(isUnique, isSpatial, isFullText bool, isVector bool, indexID string, indexCols []string, comment string) string {
```
Comment (jycor) [inline sql/sqlfmt.go:105]:
```suggestion
func GenerateCreateTableIndexDefinition(isUnique, isSpatial, isFullText, isVector bool, indexID string, indexCols []string, comment string) string {
```

File: sql/analyzer/replace_order_by_distance.go [hunk 2]
```diff
@@ -12,9 +13,9 @@ func replaceIdxOrderByDistance(ctx *sql.Context, a *Analyzer, n sql.Node, scope
 	return replaceIdxOrderByDistanceHelper(ctx, scope, n, nil)
 }
 
-func replaceIdxOrderByDistanceHelper(ctx *sql.Context, scope *plan.Scope, node sql.Node, sortNode plan.Sortable) (sql.Node, transform.TreeIdentity, error) {
+func replaceIdxOrderByDistanceHelper(ctx *sql.Context, scope *plan.Scope, node sql.Node, sortNode *plan.TopN) (sql.Node, transform.TreeIdentity, error) {
 	switch n := node.(type) {
-	case plan.Sortable:
+	case *plan.TopN:
```
Comment (jycor) [inline sql/analyzer/replace_order_by_distance.go:18]: Does `*plan.Sort` not work with vector indexes?

**Review by jycor — APPROVED** [review]
Small comments, but overall LGTM!

**Review by max-hoffman — APPROVED** [review]
LGTM, nothing in particular stands out

**Pull request merged**